### DISCUSSION TITLE
How do I actually use an NDI source?

### DISCUSSION THREAD

**Discussion by webshift1** [vcync/modV#928] [Q&A] [answered]
I just downloaded this software and want to use it with a few other pieces of visual software (synesthesia, nestdrop).  I can set up an NDI source, but I am completely lost on how to get that source to be useable in ModV.  Thanks in advance for the help!

**Comment by 2xAA** [comment, marked as answer]
Hey. So once you've set up a source, grab the Texture 2D module and add it to a group.

1. Click on the module
2. In the Module Inspector panel, select "canvas" from the first dropdown list
3. Now find the NDI group in the canvas list in the second dropdown list
4. Select your source
5. Enable the module

NDI input should now be working.

Apologies this is a bit of an awkward way to use NDI.
I have a [gigantic update to modV](https://github.com/vcync/modV/pull/921) coming soon which actually adds NDI output from modV too.

We have an issue tracking the usability of NDI (https://github.com/vcync/modV/issues/536), it's on the to-do list.

**Reply by webshift1** [reply]
Thank you so much for getting back to me.  I did finally sort this out, but now maybe someone else will find this answer useful as well
